<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!--
    Topic Remapper Launch File
    
    This launch file starts the topic_remapper node with configurable parameters.
    
    Arguments:
      - input_topic: The input topic to subscribe to (required)
      - output_topic: The output topic to publish to (default: "remapped")
      - reduction_frequency: Target output frequency in Hz (default: 10.0)
  -->

  <arg name="input_topic" description="Input topic name to remap" default="/scan"/>
  <arg name="output_topic" description="Output topic name" default="/scan_remapped"/>
  <arg name="reduction_frequency" description="Target output frequency in Hz" default="10.0"/>

  <node
    pkg="perseus_sensors"
    exec="topic_remapper"
    name="topic_remapper"
    output="screen">
    <param name="input_topic" value="$(var input_topic)"/>
    <param name="output_topic" value="$(var output_topic)"/>
    <param name="reduction_frequency" value="$(var reduction_frequency)"/>
  </node>

</launch>
